packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts

```diff
@@ -136,8 +135,15 @@ export default {
   },
 
   PartialStatement: function(partial) {
-    appendChild(this.currentElement(), partial);
-    return partial;
+    let { name, loc } = partial;
```

Comment:
No, we will throw for PartialStatement, Decorator, DecoratorBlock. Possibly more, but at least these.